python/ray/rllib/ppo/runner.py

```diff
@@ -137,33 +139,29 @@ def build_loss(obs, rets, advs, acts, plog, pvf_preds):
         self.common_policy = self.par_opt.get_common_loss()
         self.variables = ray.experimental.TensorFlowVariables(
             self.common_policy.loss, self.sess)
-        self.observation_filter = get_filter(
-            config["observation_filter"], self.preprocessor.shape)
+        obs_filter = get_filter(
+            config["observation_filter"], self.env.observation_space.shape)
+        self.sampler = SyncSampler(
+            self.env, self.common_policy, obs_filter,
+            self.config["horizon"], self.config["horizon"])
+        if not is_remote:
+            # local model needs obs_filter for compute
+            self.obs_filter = obs_filter
```

Comment:
Should we just use the obs filter in the sampler?

Reply:
Any preference on keeping the (global/master) observation_filter in the `model` or can I move it into `ppo.py`?